features/layout/sidebar-navigation/sidebar-navigation.tsx

```diff
@@ -20,6 +20,26 @@ export function SidebarNavigation() {
   const router = useRouter();
   const { isSidebarCollapsed, toggleSidebar } = useContext(NavigationContext);
   const [isMobileMenuOpen, setMobileMenuOpen] = useState(false);
+  const [isLandscape, setIsLandscape] = useState(false);
+
+  useEffect(() => {
+    if (typeof window !== "undefined") {
+      setIsLandscape(window.matchMedia("(orientation: landscape)").matches);
+
+      const mediaQuery = window.matchMedia("(orientation: landscape)");
+
+      const handleOrientationChange = (event: MediaQueryListEvent) => {
+        setIsLandscape(event.matches);
+      };
+
+      mediaQuery.addEventListener("change", handleOrientationChange);
+
+      return () => {
+        mediaQuery.removeEventListener("change", handleOrientationChange);
+      };
+    }
+  }, []);
```

Comment:
This is a valid approach but could be achieved with CSS as well. From my perspective that's preferable. Have a look at the video solution. I think it's a good watch